modules/wallet/update.go

```diff
@@ -219,6 +223,10 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 // ReceiveUpdatedUnconfirmedTransactions updates the wallet's unconfirmed
 // transaction set.
 func (w *Wallet) ReceiveUpdatedUnconfirmedTransactions(txns []types.Transaction, _ modules.ConsensusChange) {
+	if err := w.tg.Add(); err != nil {
+		panic(err)
+	}
+	defer w.tg.Done()
```

Comment:
There are a lot of bare panics already in the wallet code, should that be fixed in a later PR?  

David, do you mean that it's not necessary to check `err := w.tg.Add()` here at all?  

And why do the consensus set and transaction pool both need to update the wallet, especially if it introduces a race condition?